### DISCUSSION TITLE
Correlation between http request and response using OpenTelemery / Opentelemetry Collector

### DISCUSSION THREAD

**Discussion by neerajCL** [open-telemetry/opentelemetry-dotnet#5194] [Q&A]
How can we correlate http request and response using Opentelemetry / Opentelemetry Collector. 
What parameters we need to correlate.Can you suggest some good example or POC's that I can refer preferably in .Net/C#

**Comment by cijothomas** [comment]
Spans representing http calls (server and client), would have same traceid as long as they are part of the same request flow. Each span gets its own unique spanid as well, and the spans will have parent_span_id pointing to its parent. These are the general things facilitating correlation. Its rarely the case that one has to manually stitch these together, as most tracing backends does this automatically for you by looking at traceid,spanid,parentid etc.

Jaeger is one such popular tracing tool. Please try the tutorial https://github.com/open-telemetry/opentelemetry-dotnet/tree/main/docs/trace/getting-started-jaeger and incrementally modify the example to do more things (like outbound http calls, manual span (activity) etc.), and see the correlation in action.
